### PR TITLE
Fix empty paramValueType conversion

### PR DESCRIPTION
**Pull request by JeromeJu**
# Changes

This commit fixes the paramValueType that is not handled if input is an empty string,
which could lead to issues for 'marshal' in knative.

FIx: #5500 
/kind bug

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [n/a] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [n/a] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [n/a] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [n/a] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
NONE
```

### CODE REVIEW

File: pkg/apis/pipeline/v1beta1/param_conversion.go [hunk 1]
```diff
@@ -61,13 +69,22 @@ func (p *Param) convertFrom(ctx context.Context, source v1.Param) {
 }
 
 func (v ParamValue) convertTo(ctx context.Context, sink *v1.ParamValue) {
-	sink.Type = v1.ParamType(v.Type)
+	if v.Type != "" {
```
Comment (JeromeJu) [inline pkg/apis/pipeline/v1beta1/param_conversion.go:72]: Thanks to the discussion with @Yongxuanzhang I changed both the conversion for `paramValue` and `paramSpec`. But I was wondering in which case would we bump into the conversion of an empty string input? Thought the webhook should already taken care of that in the earlier stage?

Comment (lbernick) [reply pkg/apis/pipeline/v1beta1/param_conversion.go:72]: This can happen with resources already stored in etcd without a type set (that were created before we added defaulting logic for param values)

File: pkg/apis/pipeline/v1beta1/taskrun_conversion_test.go [hunk 2]
```diff
@@ -78,13 +78,15 @@ func TestTaskrunConversion(t *testing.T) {
 					Name: "param-task-1",
 					Value: v1beta1.ParamValue{
 						ArrayVal: []string{"value-task-1"},
+						Type:     "string",
```
Comment (lbernick) [inline pkg/apis/pipeline/v1beta1/taskrun_conversion_test.go:81]: Do you know why tests passed before you added this?

Comment (JeromeJu) [reply pkg/apis/pipeline/v1beta1/taskrun_conversion_test.go:81]: I think previously these remain empty string for `paramSpec` and `paramValue` so the round trip would pass.

Comment (JeromeJu) [reply pkg/apis/pipeline/v1beta1/taskrun_conversion_test.go:81]: And this also made wonder if we need extra tests with the empty string for paramType?

Comment (lbernick) [reply pkg/apis/pipeline/v1beta1/taskrun_conversion_test.go:81]: yes, I think we do need a new test case. I thought about it some more and I think your explanation is not quite right-- if the test input is an empty string, and the type cast is what's failing, ConvertTo would have failed. I think my original hypothesis in the bug was wrong; it's not the type cast that failed, it's that the output was type = "" which knative isn't able to marshal (serialize). Anyway your approach here is still correct, please just add the new test case.

**Review by vdemeester — APPROVED** [review]

**Comment by lbernick** [comment]
@JeromeJu it looks like you might have accidentally included a go binary-- happy to lgtm once that's removed

**Comment by JeromeJu** [comment]
> @JeromeJu it looks like you might have accidentally included a go binary-- happy to lgtm once that's removed

Thanks @lbernick Sorry that I'm not sure how my vscode generated this.

**Comment by vdemeester** [comment]
Well, I realize I did the release without this fix 😓 
So I am moving it to 0.40.0 and will cherry-pick for a 0.40.1 near the end of the week 🙃
